### PR TITLE
Add tests for podrecreation package

### PR DESCRIPTION
**Pull request by sebrandon1**
Brings the `podrecreation` package from 0% to 71.8% of statements.

Had to add `ClearTestClientsHolder` to help clean up the singleton's existing k8sclient (and corresponding generated test objects).

### CODE REVIEW

File: cnf-certification-test/lifecycle/podrecreation/podrecreation_test.go [hunk 1]
```diff
@@ -15,3 +15,120 @@
...
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/test-network-function/cnf-certification-test/internal/clientsholder"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func generatePod(name, kind string) corev1.Pod {
```
Comment (edcdavid) [inline cnf-certification-test/lifecycle/podrecreation/podrecreation_test.go:29]: rename kind to ownerKind

**Review by edcdavid — APPROVED** [review]

**Review by jmontesi — APPROVED** [review]

File: cnf-certification-test/lifecycle/podrecreation/podrecreation_test.go [hunk 2]
```diff
@@ -15,3 +15,120 @@
...
+		// Build a test clientsHolder
+		var testRuntimeObjects []runtime.Object
+		for i := range tc.testNodes {
+			x := tc.testNodes[i]
+			testRuntimeObjects = append(testRuntimeObjects, &x)
+		}
+		// Clean and recreate the clientsHolder
+		clientsholder.ClearTestClientsHolder()
+		_ = clientsholder.GetTestClientsHolder(testRuntimeObjects)
+		err := CordonHelper("node1", tc.operation)
```
Comment (greyerof) [inline cnf-certification-test/lifecycle/podrecreation/podrecreation_test.go:131]: You're only checking the err value, but why not also check whether the node has been actually cordoned/uncordoned? After the `CordonHelper()` call, you can get that node again using the k8s mocked client and check whether the node.Spec.Unschedulable is true/false depending on the tc.operation value.

Comment (sebrandon1) [reply cnf-certification-test/lifecycle/podrecreation/podrecreation_test.go:132]: Done!

File: cnf-certification-test/lifecycle/podrecreation/podrecreation_test.go (hunk 1 above)
Comment (sebrandon1) [reply cnf-certification-test/lifecycle/podrecreation/podrecreation_test.go]: Done!

**Review by greyerof — APPROVED** [review]

**Pull request merged**